In [28]:
import pandas as pd
import numpy as np

In [22]:
chart_events = pd.read_csv('./chartevents.csv')
d_items = pd.read_csv('./d_items.csv')
# icu_stays = pd.read_csv('./icu_stays.csv')
# ingredient_events = pd.read_csv('./ingredientevents.csv')
# input_events = pd.read_csv('./inputevents.csv')
# output_events = pd.read_csv('./outputevents.csv')
# procedure_events = pd.read_csv('./procedureevents.csv')



In [7]:
(d_items.category == 'Respiratory').sum()

np.int64(170)

In [11]:
d_items.head()

,itemid,label,abbreviation,linksto,category,unitname,param_type,lownormalvalue,highnormalvalue
0,220001,Problem List,Problem List,chartevents,General,NaN,Text,NaN,NaN
1,220003,ICU Admission date,ICU Admission date,datetimeevents,ADT,NaN,Date and time,NaN,NaN
2,220045,Heart Rate,HR,chartevents,Routine Vital Signs,bpm,Numeric,NaN,NaN
3,220046,Heart rate Alarm - High,HR Alarm - High,chartevents,Alarms,bpm,Numeric,NaN,NaN
4,220047,Heart Rate Alarm - Low,HR Alarm - Low,chartevents,Alarms,bpm,Numeric,NaN,NaN


### Extract column value based on another column in Pandas

#### Filtering all rows in d_items.csv's 'category' colum, keeping only the rows from 'label', abbreviation' and 'itemid' columns with the rows' values of 'Respiratory'

In [24]:
respiratory_items_id = d_items.loc[d_items['category'] == 'Respiratory', 'itemid']
respiratory_items_label = d_items.loc[d_items['category'] == 'Respiratory', 'label']
respiratory_items_abbreviation = d_items.loc[d_items['category'] == 'Respiratory', 'abbreviation']

In [36]:
len(respiratory_items_id)

170

In [40]:
respiratory_items_label

28                 Respiratory Rate
29                   Vital Capacity
34                   CO2 production
36      O2 saturation pulseoxymetry
37                       Resistance
                   ...             
4066                   Blocker Mark
4067      Blocker External Location
4068                Blocker Secured
4069         Blocker Balloon Status
4070                 Blocker Volume
Name: label, Length: 170, dtype: str

#### Constructing final data Dataframe temporarily named 'cloned_dataset'

In [88]:
cloned_dataset = pd.DataFrame(columns=respiratory_items_label)

In [105]:
len(chart_events['subject_id'].unique())

152

In [89]:
cloned_dataset

label,Respiratory Rate,Vital Capacity,CO2 production,O2 saturation pulseoxymetry,Resistance,PEEP set,Intubation - Details,O2 Flow,Inspired O2 Fraction,Airway Type,...,Vaporizer checked (Anes),Amsorb changed (Anes),Intellivent (Hamilton Vent Mode),P0.1,Blocker Type,Blocker Mark,Blocker External Location,Blocker Secured,Blocker Balloon Status,Blocker Volume


In [ ]:
cloned_dataset = cloned_dataset[0:0]
current_row_in_cloned_dataset = 0
current_subject_id = chart_events['subject_id'].unique()[current_row_in_cloned_dataset]
for current_row_in_chart_events, itemid in enumerate(chart_events['itemid']):
    # if int(current_row_in_chart_events) > 1000:
    #     break
    print(f'cloned_dataset rows: {len(cloned_dataset)}')
    print(f'current_row_in_chart_events: {current_row_in_chart_events}')
    print(f'current_row_in_cloned_dataset: {current_row_in_cloned_dataset}')
    print(f'current_subject_id: {current_subject_id}')
    print('-'*35)
    if chart_events.iloc[current_row_in_chart_events].subject_id == current_subject_id and itemid in id_list:
        cloned_dataset.loc[current_row_in_cloned_dataset, d_items.loc[d_items['itemid'] == itemid, 'label']] = chart_events.iloc[current_row_in_chart_events].valuenum
    if chart_events.iloc[current_row_in_chart_events].subject_id != current_subject_id:
        current_row_in_cloned_dataset += 1
        current_subject_id = chart_events['subject_id'].unique()[current_row_in_cloned_dataset]

In [ ]:
cloned_dataset.head()

In [81]:
len(respiratory_items_label)

170

In [41]:
dataset = pd.DataFrame(columns=respiratory_items_label)
dataset.head()

label,Respiratory Rate,Vital Capacity,CO2 production,O2 saturation pulseoxymetry,Resistance,PEEP set,Intubation - Details,O2 Flow,Inspired O2 Fraction,Airway Type,...,Vaporizer checked (Anes),Amsorb changed (Anes),Intellivent (Hamilton Vent Mode),P0.1,Blocker Type,Blocker Mark,Blocker External Location,Blocker Secured,Blocker Balloon Status,Blocker Volume


In [ ]:
# cloned_dataset.to_csv('./cloned_data.csv')

In [145]:
len(chart_events['subject_id'].unique())

152

In [146]:
len(chart_events)

1048575

#### Splitting 152 unique subject_id in chart_events.csv with each subject_id is a new csv file.

In [ ]:
# i = 0
# patient_id = 1
# subject_id_list = list(chart_events['subject_id'].unique())
# current_subject_id = subject_id_list.pop(0)
# for j in range(len(chart_events)):
#     # if j > 5000:
#     #     break
#     print(j)
#     print(patient_id)
#     print('-'*35)
#     if chart_events.iloc[j].subject_id != current_subject_id:
#         print('writing file...')
#         patient_data = chart_events.iloc[i:j]
#         print(len(patient_data))
#         patient_data.to_csv(f'../patient_data/patient_{patient_id}.csv')
#         patient_id += 1
#         i = j
#         current_subject_id = subject_id_list.pop(0)
# chart_events.iloc[1038833:].to_csv('../patient_data/patient_152.csv')
    